tools/halide_image_io.h

```diff
@@ -102,9 +106,19 @@ struct FileOpener {
             fclose(f);
         }
     }
+    // read a line of data skipping lines that begin with '#"
+    char *readLine() {
```

Comment:
This function really should take a pointer to a buffer with a size, rather than return a pointer to a member buffer.